File: app/src/main/java/ru/fitsme/android/domain/entities/returns/ReturnsOrderItem.java
package ru.fitsme.android.domain.entities.returns;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import ru.fitsme.android.domain.entities.order.OrderItem;

public class ReturnsOrderItem {
    @SerializedName("id")
    private int id;

    @SerializedName("orderitems")
    private OrderItem orderItem;

    @SerializedName("returns")
    private int returns;

    @Expose
    private String indicationNumber = "";
    @Expose
    private String cardNumber = "";

    public int getId() {
        return id;
    }

    public OrderItem getOrderItem() {
        return orderItem;
    }

    public int getReturns() {
        return returns;
    }

    public ReturnsOrderItem() {
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ReturnsOrderItem)) return false;
        ReturnsOrderItem that = (ReturnsOrderItem) o;
        return getId() == that.getId() &&
                getOrderItem() == that.getOrderItem() &&
                getReturns() == that.getReturns();
    }

    @Override
    public int hashCode() {
        int result = 17;
        int prime = 31;
        result = prime * result + (int) getId();
        result = prime * result + getOrderItem().hashCode();
        result = prime * result + getReturns();
        return result;
    }
}

File: app/src/main/java/ru/fitsme/android/domain/entities/returns/ReturnsOrderItem.kt
package ru.fitsme.android.domain.entities.returns;

import android.os.Parcelable
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import kotlinx.android.parcel.Parcelize

import ru.fitsme.android.domain.entities.order.OrderItem;

@Parcelize
data class ReturnsOrderItem(
    @SerializedName("id") val id: Int,
    @SerializedName("orderitems") val orderItem: OrderItem,
    @SerializedName("returns") val returns: Int
): Parcelable {

    constructor() : this(0, OrderItem(), 0)

    @Expose
    val indicationNumber: String = ""

    @Expose
    val cardNumber: String = ""
}
